### PR TITLE
Fix panic in handling global variables

### PR DESCRIPTION
**Pull request by yuxincs**
This PR fixes two panics due to the assumption that the RHS of the global variable assignments are always variables, which isn't true when RHS is a constant (either defined in the current package, or referenced from upstream package)

We also take this opportunity to refactor the code a little bit to be more readable and maintainable.

Depends on #68

### CODE REVIEW

File: testdata/src/go.uber.org/globalvars/globalvarinit.go [hunk 1]
```diff
@@ -94,3 +96,14 @@ func foo() *int {
 	print(multiNonNil, multiNil, nonnilMethod, assignedNilable)
 	return nil
 }
+
+// Now we test a corner case where a constant is assigned to a global variable.
```
Comment (sonalmahajan15) [inline testdata/src/go.uber.org/globalvars/globalvarinit.go:100]:
```suggestion
// Below test checks when a constant is assigned to a global variable.
```

Comment (sonalmahajan15) [reply testdata/src/go.uber.org/globalvars/globalvarinit.go:100]: Feel free to ignore this suggestion. 🙂  My reasoning was that the comment with "Now" makes sense in the context of this PR, but the story might be lost later in code.

**Review by sonalmahajan15 — APPROVED** [review]
LGTM! Just a minor suggestion, feel free to ignore :)

Btw curious to know what was the panic here?

File: testdata/src/go.uber.org/globalvars/globalvarinit.go (hunk 1 above)
Comment (yuxincs) [reply testdata/src/go.uber.org/globalvars/globalvarinit.go:100]: I agree, good catch! updated!
